plasmapy/analysis/tests/test_fit_functions.py

```diff
@@ -0,0 +1,497 @@
+            params[0] = "hello"
+            foo.func(5, *params)
+
+    def test_func_err(self):
+        foo = self.ff_class(
+            params=self._test_params, param_errors=self._test_param_errors
+        )
+
+        for x in (0, 1.0, np.linspace(10, 30, num=20)):
+            assert np.allclose(
+                foo.func_err(x),
+                self.func_err(x, self._test_params, self._test_param_errors),
+            )
+
+        x = [4, 5, 6]
+        results = foo.func_err(x, x_err=0.1, rety=True)
+        assert np.allclose(
+            results[0],
+            self.func_err(
+                np.array(x), self._test_params, self._test_param_errors, x_err=0.1
+            ),
+        )
+        assert np.allclose(results[1], self.func(np.array(x), *self._test_params))
+
+        with pytest.raises(ValueError):
+            foo.func_err("hello")
+
+        with pytest.raises(ValueError):
+            foo.func_err(5, x_err="goodbye")
+
+        with pytest.raises(ValueError):
+            foo.func_err(5, x_err=[0.1, 0.1])
+
+    def test_call(self):
+        foo = self.ff_class()
+        foo.params = self._test_params
+        foo.param_errors = self._test_param_errors
+
+        for x in (0, 1.0, np.linspace(10, 30, num=20)):
+            assert np.allclose(foo(x), self.func(x, *self._test_params))
+
+            # also return error
+            y, y_err = foo(x, reterr=True)
+            assert np.allclose(y, self.func(x, *self._test_params))
+            assert np.allclose(
+                y_err, self.func_err(x, self._test_params, self._test_param_errors),
+            )
+
+        x = [4, 5, 6]
+        x_err = 0.05
+        assert np.allclose(foo(x), self.func(np.array(x), *self._test_params))
+        y, y_err = foo(x, x_err=x_err, reterr=True)
+        assert np.allclose(y, self.func(np.array(x), *self._test_params))
+        assert np.allclose(
+            y_err,
+            self.func_err(
+                np.array(x), self._test_params, self._test_param_errors, x_err=x_err
+            ),
+        )
+
+        with pytest.raises(ValueError):
+            foo("hello")
+
+        with pytest.raises(ValueError):
+            foo(5, x_err=[1, 2], reterr=True)
+
+    @abstractmethod
+    def test_root_solve(self):
+        raise NotImplementedError
+
+    def test_curve_fit(self):
+        foo = self.ff_class()
+
+        xdata = np.linspace(-10, 10)
+        ydata = self.func(xdata, *self._test_params)
+
+        assert foo.params is None
+        assert foo.param_errors is None
+        assert foo.rsq is None
+        assert foo.curve_fit_results is None
+
+        foo.curve_fit(xdata, ydata)
+
+        assert foo.curve_fit_results is not None
+        assert np.isclose(foo.rsq, 1.0)
+        assert np.allclose(
+            foo.param_errors, tuple([0] * len(foo.param_names)), atol=1.5e-8,
+        )
+        assert np.allclose(foo.params, self._test_params)
+
+    # TODO: test for __call__
```

Comment:
It would probably be easier in the long run to put this test in now rather than later.

Reply:
I already have.  I just forgot to delete this TODO.

Reply:
Suggestion:  Add
```Python
    # TODO: Delete this TODO
```
:cat2: